### PR TITLE
Replace resize with hook

### PR DESCRIPTION
**Pull request by pietruszka**
Follow-up to https://github.com/Scrivito/scrivito_example_app_js/pull/349#discussion_r515858241

https://www.npmjs.com/package/use-resize-observer#the-onresize-callback
> By the default the hook will trigger a re-render on all changes to the target element's width and / or height.

That's why we don't have to compare anymore an old and a new size.

### CODE REVIEW

**Comment by pietruszka** [comment]
@dcsaszar Please review

File: src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js [hunk 1]
```diff
@@ -109,6 +63,20 @@ function StaticGoogleMap({ address, apiKey, zoom, children }) {
...
+  let newHeight = height;
+
+  if (newWidth > maxWidth) {
+    newWidth = maxWidth;
+
+    const factor = height / width;
+    newHeight = Math.round(maxWidth * factor);
+  }
+
+  return { width: newWidth, height: newHeight };
```
Comment (apepper) [inline src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js:77]: Now that this is a dedicated method we can simplify it further:

```suggestion
  if (width > maxWidth) {
    const factor = height / width;
    const adjustedHeight = Math.round(maxWidth * factor);

    return { width: maxWidth, height: adjustedHeight };
  }

  return { width, height };
```

File: src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js [hunk 2]
```diff
@@ -109,6 +63,17 @@ function StaticGoogleMap({ address, apiKey, zoom, children }) {
   );
 }
 
+function scaleMapSize(width, height) {
+  if (width > maxWidth) {
```
Comment (dcsaszar) [inline src/Widgets/GoogleMapsWidget/GoogleMapsWidgetComponent.js:67]: * [ ] FYI the input can be `null` according to line 43. This function works by change with `null` input, but we may want to make it more explicit. Also, with `0` input the function may output `NaN` values. I think with a guard like `!height || !width` we would avoid any potential headache.

**Review by dcsaszar — APPROVED** [review]
LGTM. See comment.

**Review by dcsaszar — APPROVED** [review]

**Review by apepper — COMMENTED** [review]
FEEDBACK (a696188)

Nice work 🚀 ! Almost there!

I found two places that could be improved.

**Review by apepper — APPROVED** [review]
LGTM (897bb51)

Good job 🚀 !

**Pull request merged**